### PR TITLE
KAITO P1 - (AI Model Chat Feature)

### PR DESCRIPTION
**Pull request by tejhan**
## KAITO - AI Model Chat Feature

 - Allows user to test the inference server of successfully deployed models
 - Allows user to provide prompts and adjust arguments to query the model
 - Added health pre-checks for kaito operations
 - Several adjustments & slight additions to UI
 - Fixes issues: 
 #1064, 
 #1065 
#1066 
 #1067 
 #1068 
#1069

Most recent .vsix -> [vscode-aks-tools-1.5.2-p1-12-3.zip](https://github.com/user-attachments/files/17996916/vscode-aks-tools-1.5.2-p1-12-3.zip)

 </img>
</img>

(Note: This PR builds on top of #1048 )

### CODE REVIEW

**Comment by tejhan** [comment]
.vsix for testing -> [vscode-aks-tools-1.5.0-p1-full.zip](https://github.com/user-attachments/files/17706713/vscode-aks-tools-1.5.0-p1-full.zip)

**Comment by tejhan** [comment]
Most Recent vsix -> [vscode-aks-tools-1.5.0-p1-f1.zip](https://github.com/user-attachments/files/17710367/vscode-aks-tools-1.5.0-p1-f1.zip)

**Comment by tejhan** [comment]
Most recent .vsix -> [vscode-aks-tools-1.5.0-p1-11-12.zip](https://github.com/user-attachments/files/17720897/vscode-aks-tools-1.5.0-p1-11-12.zip)

File: src/commands/aksKaito/aksKaitoTest.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        return;
+    }
+
+    const { name: clusterName, armId, subscriptionId, resourceGroupName } = clusterNode.result;
+    const clusterInfo = await getKubernetesClusterInfo(sessionProvider.result, target, cloudExplorer, clusterExplorer);
+    if (failed(clusterInfo)) {
+        vscode.window.showErrorMessage(clusterInfo.error);
+        return;
+    }
+    const kubeConfigFile = await tmpfile.createTempFile(clusterInfo.result.kubeconfigYaml, "yaml");
```
Comment (hsubramanianaks) [inline src/commands/aksKaito/aksKaitoTest.ts:58]: do we need tempFileCreation at every page for Kaito? isn't it check if it exists already and if it doesn't exist create it? I would say let's do all this refactor in another PR, just note it down. Thank you.

File: webview-ui/src/KaitoTest/KaitoTest.module.css [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    width: 16rem;
+    height: 0.75rem;
+    cursor: pointer;
+    background: none;
+    border-radius: 5px;
+    outline: none;
+}
+
+/* Track styling */
+.main input[type="range"]::-webkit-slider-runnable-track {
```
Comment (hsubramanianaks) [inline webview-ui/src/KaitoTest/KaitoTest.module.css:50]: @tejhan can we please test this in all platforms so that the slider works fine.

File: src/panels/KaitoTestPanel.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                "port-forward",
+                `svc/workspace-${this.modelName}`,
+                `${localPort}:80`,
+                "--kubeconfig",
+                this.kubeConfigFilePath,
+            ]);
+
+            // slight delay to allow for port forwarding to initialize
+            await new Promise((resolve) => setTimeout(resolve, 2000));
+            const curlCommand = `curl -X POST http://localhost:${localPort}/chat -H "accept: application/json" -H \
```
Comment (hsubramanianaks) [inline src/panels/KaitoTestPanel.ts:120]: In windows it would be curl.exe and let's plan to test this different platforms (windows/linux/mac)

File: src/panels/KaitoTestPanel.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        maxLength: number,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
+        // prevents user from re-submitting while a query is in progress
+        if (this.sendingQuery) {
+            return;
+        }
+        await longRunning(`Sending query...`, async () => {
+            this.sendingQuery = true;
+            const localPort = await getPort();
```
Comment (hsubramanianaks) [inline src/panels/KaitoTestPanel.ts:109]: getPort is library that gives you random port? can you please leave comment here for other folks to know.

File: webview-ui/src/manualTest/kaitoTestTests.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+export function getKaitoTestScenarios() {
+    const initialState: InitialState = {
+        clusterName: "ai-service",
+        modelName: "falcon-7b-instruct",
+        output: "What is the meaning of life?\n\nThe meaning of life is to be happy.",
+    };
+
+    function getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        void webview;
```
Comment (hsubramanianaks) [inline webview-ui/src/manualTest/kaitoTestTests.tsx:19]: this void webview is required? can you please remove it if not needed. Thank you.

File: src/panels/KaitoTestPanel.ts [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+            output: "",
+        };
+    }
+    getTelemetryDefinition(): TelemetryDefinition<"kaitoTest"> {
+        return {
+            queryRequest: true,
+        };
+    }
+    getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        void webview;
```
Comment (hsubramanianaks) [inline src/panels/KaitoTestPanel.ts:57]: same suggestion here: please comment why it is needed or remove it. Thank you.

**Comment by tejhan** [comment]
Replaced by PR #1106
